src/syft_rds/models/runtime_models.py

```diff
@@ -1,21 +1,168 @@
+        return value
+
+
+class DockerMount(BaseModel):
+    source: PathLike
+    target: PathLike
+    mode: Literal["ro", "rw"] = "ro"
+
+    @field_validator("source")
+    def validate_source_path_exists(cls, value: PathLike) -> PathLike:
+        source_path = Path(value).expanduser().resolve()
+        if not source_path.exists():
+            # raise FileNotFoundError(f"Source path '{value}' does not exist")
+            logger.warning(f"Source path '{value}' does not exist")
+        return value
+
+
+class DockerRuntimeConfig(BaseRuntimeConfig):
+    dockerfile: PathLike | None = Field(default=None, exclude=True)
+    entrypoint_script: PathLike | None = Field(default=None, exclude=True)
+    dockerfile_content: str
+    image_name: str | None = None
+    cmd: list[str] = ["python"]
+    app_name: str | None = None
+    extra_mounts: list[DockerMount] = Field(default_factory=list)
+
+    @model_validator(mode="before")
+    @classmethod
+    def load_content_from_path(cls, values: dict[str, Any]) -> dict[str, Any]:
+        """
+        If a 'dockerfile' path is provided, read its content into 'dockerfile_content'.
+        This validator runs before any other field validation.
+        """
+        if "dockerfile" in values and values["dockerfile"] is not None:
+            dockerfile_path = Path(values["dockerfile"]).expanduser().resolve()
+            if not dockerfile_path.exists():
+                raise FileNotFoundError(
+                    f"The specified Dockerfile does not exist at: {dockerfile_path}"
+                )
+            values["dockerfile_content"] = dockerfile_path.read_text()
+        elif "dockerfile_content" not in values:
+            raise ValueError(
+                "You must provide a path to a Dockerfile via the 'dockerfile' argument."
+            )
+        return values
+
+    @field_validator("dockerfile_content")
+    def validate_dockerfile_content(cls, dockerfile_content: str) -> str:
+        if not dockerfile_content:
+            raise ValueError("Dockerfile cannot be empty")
+        return dockerfile_content.strip()
+
+    def __hash__(self) -> int:
+        return hash(self.dockerfile_content)
+
+    def __eq__(self, __value: object) -> bool:
+        if not isinstance(__value, DockerRuntimeConfig):
+            return False
+        return self.dockerfile_content == __value.dockerfile_content
+
+
+class KubernetesRuntimeConfig(BaseRuntimeConfig):
+    image: str
+    namespace: str = "syft-rds"
+    num_workers: int = 1
+    cmd: list[str] = Field(default_factory=list)
+
+
+RuntimeConfig = PythonRuntimeConfig | DockerRuntimeConfig | KubernetesRuntimeConfig
+
+
 class Runtime(ItemBase):
     __schema_name__ = "runtime"
+    __table_extra_fields__ = [
+        "name",
+        "kind",
+    ]
 
-    name: str
-    description: str
+    name: str | None = None
+    kind: RuntimeKind
+    config: RuntimeConfig = Field(default_factory=dict)
```

Comment:
This might introduce a bug (but not sure). Can you check what happens here if we create a Runtime with DockerRuntimeConfig, serialize (ie, write to store) and deserialize (read from store)? 

I suspect it forgets the subtype and always returns a `RuntimeConfig` (instead of the DockerRuntimeConfig we saved)